packages/lib/response/service.ts

```diff
@@ -80,85 +82,110 @@ const responseSelection = {
+        });
+
+        if (!responsePrisma) {
+          throw new ResourceNotFoundError("Response from PersonId", personId);
+        }
+
+        let responses: Array<TResponse> = [];
+
+        responsePrisma.forEach((response) => {
+          responses.push({
+            ...response,
+            person: response.person ? transformPrismaPerson(response.person) : null,
+            tags: response.tags.map((tagPrisma: { tag: TTag }) => tagPrisma.tag),
+          });
+        });
+
+        return responses;
+      } catch (error) {
+        if (error instanceof Prisma.PrismaClientKnownRequestError) {
+          throw new DatabaseError("Database operation failed");
+        }
+
+        throw error;
+      }
+    },
+    [`getResponsesByPersonId-${personId}-${page}`],
+    {
+      tags: [responseCache.tag.byPersonId(personId)],
+      revalidate: SERVICES_REVALIDATION_INTERVAL,
     }
+  )();
 
-    throw error;
-  }
+  return responses.map((response) => ({
+    ...response,
+    ...formatResponseDateFields(response),
+  }));
 };
 
 export const getResponseBySingleUseId = async (
   surveyId: string,
-  singleUseId?: string
+  singleUseId: string
 ): Promise<TResponse | null> => {
-  validateInputs([surveyId, ZId], [singleUseId, ZString]);
-
-  try {
-    if (!singleUseId) {
-      return null;
-    }
-    const responsePrisma = await prisma.response.findUnique({
-      where: {
-        surveyId_singleUseId: { surveyId, singleUseId },
-      },
-      select: responseSelection,
-    });
-
-    if (!responsePrisma) {
-      return null;
-    }
-
-    const response: TResponse = {
-      ...responsePrisma,
-      person: responsePrisma.person ? transformPrismaPerson(responsePrisma.person) : null,
-      tags: responsePrisma.tags.map((tagPrisma: { tag: TTag }) => tagPrisma.tag),
-    };
-
-    return response;
-  } catch (error) {
-    if (error instanceof Prisma.PrismaClientKnownRequestError) {
-      throw new DatabaseError("Database operation failed");
+  const response = await unstable_cache(
+    async () => {
+      validateInputs([surveyId, ZId], [singleUseId, ZString]);
+
+      try {
+        const responsePrisma = await prisma.response.findUnique({
+          where: {
+            surveyId_singleUseId: { surveyId, singleUseId },
+          },
+          select: responseSelection,
+        });
+
+        if (!responsePrisma) {
+          return null;
+        }
+
+        const response: TResponse = {
+          ...responsePrisma,
+          person: responsePrisma.person ? transformPrismaPerson(responsePrisma.person) : null,
+          tags: responsePrisma.tags.map((tagPrisma: { tag: TTag }) => tagPrisma.tag),
+        };
+
+        return response;
+      } catch (error) {
+        if (error instanceof Prisma.PrismaClientKnownRequestError) {
+          throw new DatabaseError("Database operation failed");
+        }
+
+        throw error;
+      }
+    },
+    [`getResponseBySingleUseId-${surveyId}-singleuse-${singleUseId}`],
```

Comment:
I would prefer if we just concat the arguments:
`getResponseBySingleUseId-${surveyId}-${singleUseId}`
It's easier to follow and read.

Reply:
Done